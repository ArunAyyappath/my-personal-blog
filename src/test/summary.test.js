import React from "react";
import SummaryCardCompo from "../component/summaryCard";
import { configure, shallow, cleanup, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });

afterEach = () => {
  cleanup;
};

describe("<SummaryCardCompo/>", () => {
  it("chek snapshot of the component", () => {
    const wrapper = renderer
      .create(
        <SummaryCardCompo
          key={Date.now()}
          titleHead={"Reader"}
          dataInner={"<p>Loading</p>"}
          imagePath={""}
        />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("check the prop which is passed to", () => {
    const wrapper = mount(
      <SummaryCardCompo
        key={Date.now()}
        titleHead={"Reader"}
        dataInner={"<p>Loading</p>"}
        imagePath={""}
      />
    );
    expect(wrapper.find("h2").text()).toBe("Reader");
  });
});
