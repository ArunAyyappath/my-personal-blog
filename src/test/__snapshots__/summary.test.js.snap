// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryCardCompo/> Compare snapshot of the component 1`] = `
<div
  style={
    Object {
      "float": "left",
      "margin": "50px 40px 10px 60px",
      "width": "300px",
    }
  }
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
  >
    <button
      className="MuiButtonBase-root MuiCardActionArea-root"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <img
        alt="Reader"
        className="MuiCardMedia-root MuiCardMedia-media"
        height="140"
        src="blog1N.jpg"
        title="Light Up Your Thoughts"
      />
      <div
        className="MuiCardContent-root"
      >
        <h6
          className="MuiTypography-root makeStyles-font-3 MuiTypography-h6 MuiTypography-gutterBottom"
        />
        <h2
          className="MuiTypography-root makeStyles-font-3 MuiTypography-h5 MuiTypography-gutterBottom"
        >
          Reader
        </h2>
        <div
          className="MuiTypography-root makeStyles-align-2 MuiTypography-body2 MuiTypography-colorTextSecondary"
        >
          Loading
        </div>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight"
      />
    </button>
    <div
      className="MuiCardActions-root MuiCardActions-spacing"
    />
  </div>
</div>
`;
